### PR TITLE
feat(statedb): Per block logs journal

### PR DESCRIPTION
**Pull request by calbera**
- Allow the logs journal to store the entire block's logs, similar to Geth

### CODE REVIEW

File: eth/core/processor.go [hunk 1]
```diff
@@ -242,20 +235,16 @@ func (sp *StateProcessor) Finalize(
...
-	var logs []*types.Log
-	for _, receipt := range sp.receipts {
-		receipt.BlockHash = blockHash
-		for _, log := range receipt.Logs {
-			log.BlockHash = blockHash
-		}
-		logs = append(logs, receipt.Logs...)
+
+	// Set the missing fields on receipts effective gas price and block hash.
+	if err := sp.receipts.DeriveFields(
```
[the remark this answers is not in the document]
Comment (calbera) [reply eth/core/processor.go:240]: but we iterate through the receipts -> to logs either way

[the remark this answers is not in the document]
Comment (itsdevbear) [reply eth/core/processor.go:240]: I would mirror geth.

File: eth/core/state/interfaces.go [hunk 2]
```diff
@@ -101,6 +101,8 @@ type (
 		Logs() []*coretypes.Log
 		// GetLogs returns the logs of the tx with the given metadata.
 		GetLogs(hash common.Hash, blockNumber uint64, blockHash common.Hash) []*coretypes.Log
+		// GetBlockLogsAndClear returns the logs of the block and clears the journal.
+		GetBlockLogsAndClear(blockHash common.Hash) []*coretypes.Log
```
Comment (itsdevbear) [inline eth/core/state/interfaces.go:105]: we need to stop adding shit not in geth keke

Comment (calbera) [reply eth/core/state/interfaces.go:105]: better separation of concerns to have the block logs list in the logs journal imo, especially based on our abstractions. geth stores logs per block as well

Comment (itsdevbear) [reply eth/core/state/interfaces.go:105]: why do we need an Clear then?

just clear on StateDB Reset

Comment (itsdevbear) [reply eth/core/state/interfaces.go:105]: That's how Geth does it tho lmeow

Comment (calbera) [reply eth/core/state/interfaces.go:105]: lmao wow geth never clears the logs map

Comment (calbera) [reply eth/core/state/interfaces.go:105]: we can also not clear but I think its a good optimization

**Comment by itsdevbear** [comment]
Mirror geth good project for now. Optmize later for sure

**Comment by calbera** [comment]
working off of #734, should be merged only after that PR is in

**Comment by calbera** [comment]
Think its worth shelving this for now. Storing logs per block is definitely more inefficient than what we have currently and does not add any new functionality. This change is more of a refactor that we can avoid for now.

will address the receipts storage issue in another PR.

**Pull request closed without merging**